search/src/main/java/com/jjswigut/search/presentation/LocationHelper.kt

```diff
@@ -0,0 +1,92 @@
+package com.jjswigut.search.presentation
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import android.content.pm.PackageManager
+import android.location.Location
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.core.app.ActivityCompat.requestPermissions
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+
+class LocationHelper {
```

Comment:
So I think the way its written, if lastLocation works, no location updates are needed, if it doesn't, it requests the current location updates, but then on line 90 it removes the location updates. It shouldn't be a constant thing unless I'm missing something?

Reply:
if it does get a location update yeah it would stop requesting it. but usually the act of fetching for a location is tied to a screen in your app so if that screen left before location returned since it is an async operation to get location, you should also remove updates when that screen is no longer at the UI forefront